src/scheduler/server_info.c

```diff
@@ -388,9 +389,28 @@ query_server(status *pol, int pbs_sd)
 			update_counts_on_run(cts, sinfo->running_jobs[i]->resreq);
 
 			update_counts_on_run(allcts, sinfo->running_jobs[i]->resreq);
+			/* Since we are already looping on running jobs, associcate running
+			 * subjobs to their parent.
+			 */
+			if ((sinfo->running_jobs[i]->job->is_subjob) &&
+			    (1 == associate_array_parent(sinfo->running_jobs[i], sinfo))) {
+				sinfo->fairshare = NULL;
+				free_server(sinfo);
+				return NULL;
+			}
 		}
 		create_total_counts(sinfo, NULL, NULL, SERVER);
 	}
+	if (job_arrays_associated == FALSE) {
+		for (i = 0; sinfo->running_jobs[i] != NULL; i++) {
+			if ((sinfo->running_jobs[i]->job->is_subjob) &&
+			    (1 == associate_array_parent(sinfo->running_jobs[i], sinfo))) {
+				sinfo->fairshare = NULL;
+				free_server(sinfo);
+				return NULL;
+			}
+		}
+	}
```

Comment:
The reason being that we already run a loop on running jobs when a run limit is set. I wanted to avoid looping on running jobs again. But then for cases when run limits are not set, we need to loop on running jobs anyways and that's why the second loop. It looks vague but saves us looping on running jobs in some cases.